.github/workflows/test-suite.yml

```diff
@@ -257,89 +324,65 @@ jobs:
   execution-engine-integration-ubuntu:
     name: execution-engine-integration-ubuntu
     runs-on: ubuntu-latest
-    needs: cargo-fmt
     steps:
     - uses: actions/checkout@v3
-    - uses: actions/setup-go@v3
-      with:
-        go-version: '1.20'
-    - uses: actions/setup-dotnet@v3
-      with:
-        dotnet-version: '6.0.201'
     - name: Get latest version of stable Rust
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      with:
+          channel: stable
+          cache-target: release
+          components: rustfmt
+          cache: false
     - name: Run exec engine integration tests in release
       run: make test-exec-engine
-  check-benchmarks:
-    name: check-benchmarks
+  check-code:
+    name: check-code
     runs-on: ubuntu-latest
-    needs: cargo-fmt
     steps:
     - uses: actions/checkout@v3
     - name: Get latest version of stable Rust
-      run: rustup update stable
-    - name: Typecheck benchmark code without running it
-      run: make check-benches
-  clippy:
-    name: clippy
-    runs-on: ubuntu-latest
-    needs: cargo-fmt
-    steps:
-    - uses: actions/checkout@v3
-    - name: Get latest version of stable Rust
-      run: rustup update stable
+      uses: moonrepo/setup-rust@v1
+      env:
+          CARGO_INCREMENTAL: 1
```

Comment:
Is this intended for this step? Not sure if it compiles anything here

Reply:
My thinking was that at each step we compile lighthouse and its deps to do the all the various checks. I thought incremental compilation would be beneficial here. This isn't the case? @jimmygchen

Reply:
Oh it makes sense - in that case i think we need to shift this `env` one level up to the _job_ level, e.g. 

```
  check-code:
    name: check-code
    runs-on: ubuntu-latest
    env:
        CARGO_INCREMENTAL: 1
```

At the moment it's scoped at the _step_ level, which i don't think is available to other steps under this job.